api/models/site-build-task.js

```diff
@@ -8,6 +11,62 @@ const associate = ({ BuildTaskType, Site, SiteBuildTask }) => {
   });
 };
 
+async function startBuildTask(buildId) {
+  const siteBuildTask = this;
+  const {
+    BuildTask,
+    BuildTaskType,
+    Build,
+    Site,
+  } = siteBuildTask.sequelize.models;
+
+  const buildTask = await BuildTask.create({
+    buildTaskTypeId: siteBuildTask.buildTaskTypeId,
+    buildId,
+    name: `build: ${buildId}, type: ${siteBuildTask.buildTaskTypeId}`,
+  });
+
+  const createdBuildTask = await BuildTask.findByPk(buildTask.id, {
+    include: [
+      { model: BuildTaskType, required: true },
+      { model: Build, required: true, include: [{ model: Site, required: true }] },
+    ],
+  });
+
+  BuildTaskQueue.sendTaskMessage(createdBuildTask);
+}
+
+async function findTasksToStart({ build, startsWhen }) {
+  const {
+    BuildTaskType,
+  } = this.sequelize.models;
+  return this.findAll({
+    where: {
+      [Op.and]: [
+        {
+          [Op.or]: [
+            { branch: build.branch },
+            { branch: null },
+          ],
+        },
+        {
+          siteId: build.site,
+        },
+      ],
+    },
+    include: {
+      model: BuildTaskType,
+      where: { metadata: { startsWhen } },
+      required: true,
+    },
+  });
+}
+
+async function startBuildTasks({ build, startsWhen }) {
+  const tasks = await this.findTasksToStart({ build, startsWhen });
+  await Promise.allSettled(tasks.map(sbt => sbt.startBuildTask(build.id)));
```

Comment:
For tasks that reject, we may want to think about how we handle them in the future. If they were to fail it would be a database or queue connection problem which means its probably a larger event happening.